llm/validation.go

```diff
@@ -0,0 +1,125 @@
+// Validation functions
+
+package llm
+
+import (
+	"fmt"
+	"log/slog"
+	"slices"
+
+	"github.com/ollama/ollama/discover"
+)
+
+// Interface for GGML functionality needed by validation
+type GGMLModel interface {
+	KV() KV
+}
+
+// ValidateFlashAttentionSupport checks if flash attention is supported by the model and hardware
+func ValidateFlashAttentionSupport(ggml GGMLModel, gpus discover.GpuInfoList, flashAttnRequested bool) FlashAttentionSupport {
```

Comment:
Thanks for pulling this out.

It would be best to reduce this to something similar to:

```suggestion
func supportsFlashAttention(ggml *GGMLModel) bool {
```

and a similar function on the GPU side in the `discover` package

Reply:
Roger that, I've re-factored this now.